### PR TITLE
Update docs for HttpsEnforcerMiddleware

### PR DESCRIPTION
**Pull request by markstory**
Remove outdated information for SecurityComponent as well.

### CODE REVIEW

File: en/controllers/components/security.rst [hunk 1]
```diff
@@ -169,51 +144,14 @@ want and the Security Component will enforce them on its startup::
         {
             parent::beforeFilter($event);
 
-            if ($this->request->getParam('admin')) {
-                $this->Security->requireSecure();
+            if ($this->request->getParam('prefix') === 'admin') {
+                $this->Security->setConfig('validatePost', false);
```
Comment (saeideng) [inline en/controllers/components/security.rst]: most cases in admin area `validatePost` should be `true` as is by default 
could you change example ?

File: en/controllers/middleware.rst [hunk 2]
```diff
@@ -446,6 +447,42 @@ an option. You can also define your own parsers::
...
+
+If you want your application to only be available via HTTPS connections you can
+use the ``HttpsEnforcerMiddleware``::
+
+    use Cake\Http\Middleware\HttpsEnforcerMiddleware;
+
+    // GET requests will be automatically redirect to HTTPS if they
+    // are created with HTTP.
+    $https = new HttpsMiddleware([
+        'redirect' => true,
```
Comment (garas) [inline en/controllers/middleware.rst]: `redirect` defaults to `true`. 

Examples with non-default `redirect/statusCode/disableOnDebug` could be better.

File: en/controllers/components/security.rst (hunk 1 above)
Comment (ADmad) [reply en/controllers/components/security.rst]: Not really, I turn off SecurityComponent (actually just don't load it) for admin area :slightly_smiling_face:.

File: en/controllers/middleware.rst [hunk 3]
```diff
@@ -446,6 +447,42 @@ an option. You can also define your own parsers::
         return Csv::parse($body);
     });
 
+.. _https-enforcer-middleware:
+
+HTTPs Enforcer Middleware
```
Comment (ADmad) [inline en/controllers/middleware.rst]: Capital "S"?

File: en/controllers/components/security.rst (hunk 1 above)
Comment (markstory) [reply en/controllers/components/security.rst]: There aren't many options left to turn off. We could remove this section entirely.

Comment (saeideng) [reply en/controllers/components/security.rst]: for example for requests from other sites/server to very(payment)/... or other something
@ADmad if you was only admin yes you can remove whole security features :smile:

File: en/appendices/4-0-migration-guide.rst [hunk 4]
```diff
@@ -107,6 +107,8 @@ Component
 * ``Cake\Controller\Component\PagingComponent`` now sets paging params info as
    request attribute instead of request param. Hence you should now use
   ``$request->getAttribute('paging')`` instead of ``$request->getParam('paging')``.
+* The ``requireSecure`` behavior of ``SecurityComponent`` was removed. Instead
```
Comment (ADmad) [inline en/appendices/4-0-migration-guide.rst]: The feature isn't still removed from SecurityComponent. Wouldn't we be marking it as deprecated instead?

Comment (markstory) [reply en/appendices/4-0-migration-guide.rst]: Ooops. I'll update this note.

**Pull request merged**